crates/codegen/grammar/src/parser_definition.rs

```diff
@@ -40,16 +40,16 @@ pub enum ParserDefinitionNode {
     Versioned(Box<Self>, Vec<VersionQualityRange>),
     Optional(Box<Self>),
     ZeroOrMore(Box<Self>),
-    OneOrMore(Box<Self>),
-    Sequence(Vec<Self>),
-    Choice(Vec<Self>),
+    OneOrMore(String, Box<Self>),
```

Comment:
I suggest creating a `struct` with named fields for all these `(String, Self)` types. I think having `.name` and `.value` (or `.node`, etc...) would make it easier to understand/read through many of these callsites over `.0` and `.1`.

Reply:
Same suggestion for `NamedNode`.

Reply:
I don't mind tuples with `.0` etc for local cases, but I agree with @OmarTawfik when it comes to anything non local. I prefer `OneOrMore { name: ... }` in that case.

Reply:
Agreed on the non-local tuple and confusing `.0`, `.1` accesses in that context.

I introduced a `Named` struct in https://github.com/NomicFoundation/slang/pull/710/commits/9dbd411088c94ffe94eb5004ab5666617acd05a4, which is a named `{ name: String, node: Self `} aggregate. In this case, the node has to be generic (`T`), since it's used with `Self` in different places (in Vec, Boxed) and I didn't want to introduce a double indirection (i.e. `Vec<Box<Self>>`; note that we always have to heap-allocate due to recursive definition).

Due to the concise wrapper name, it reads surprisingly good when used with the outer enum variants and is clearer.

Let me know if that's okay.

Reply:
`pub type NamedNode = (String, Node);` flows pretty nicely in this PR - this is a single instance (not multiple variants like ParserDefinitionNode), there are no .0/.1 accesses and the types are distinct and somewhat obvious String + Node is pretty self explanatory, I'm trying to give them meaningful names already when destructuring.

Do you want me to still introduce a NamedNode as a struct, as well??

Reply:
I've introduced it in https://github.com/NomicFoundation/slang/pull/710/commits/77947ff4afe3abc0494a712baaf85f3c328c95b4 - it makes some access patterns slightly nicer but the pattern matching on node kinds got unwieldy and less readable. Let me know what you think!